js/experiment.js

```diff
@@ -1,14 +1,108 @@
-function drawRect() {
-	var c = document.getElementById("theCanvas");
-	var ctx = c.getContext("2d");
-	ctx.fillStyle = "#FF00FF";
-	ctx.fillRect(0, 0, 80, 100);
+var nodeId = 0;
+var colours = { 
+     "red": "#D77546", 
+     "green":"#2E8B57", 
+     "blue":"#437699",
+     "purple":"#46364A"
+};
+var nodeColour = colours["red"];
+//var nodes = [];
+var mode = '';
+var xmlns = 'http://www.w3.org/2000/svg';
+
+function setupSVGCanvas() {
+	'use-strict';
+
+	var svg = document.createElementNS(xmlns, 'svg');
+	svg.setAttribute('id', 'mySVG');
+	svg.setAttribute('style', 'border: 2px solid black');
+	svg.setAttribute('width', '800');
+	svg.setAttribute('height', '500');
+	svg.setAttributeNS('http://www.w3.org/2000/xmlns/', 'xmlns:xlink', 'http://www.w3.org/1999/xlink');
+	document.body.appendChild(svg);
```

Comment:
The indentation in here and some other places in your file seems to use tabs; please replace them with spaces.